src/set_theory/surreal.lean

```diff
@@ -286,12 +454,219 @@ instance : ordered_add_comm_group surreal :=
   le_antisymm       := by { rintros ⟨_⟩ ⟨_⟩ h₁ h₂, exact quotient.sound ⟨h₁, h₂⟩ },
   add_le_add_left   := by { rintros ⟨_⟩ ⟨_⟩ hx ⟨_⟩, exact pgame.add_le_add_left hx } }
 
+instance : add_comm_group surreal := by apply_instance
+
```

Comment:
This presumably was just added in testing, and can be removed?

Reply:
Ah, no, this one one is weird. 

If I do not explicitly say this here then Lean infers the `add_comm` instance from `noncomputable instance : linear_ordered_add_comm_group surreal` instead of `instance : ordered_add_comm_group` and forces me to make a lot of theorems afterwards unnecessarily `noncomputable`.

Reply:
is lowering the instance priority a good option?

Reply:
I can also just move the `noncomputable instance : linear_ordered_add_comm_group surreal` to the very end of the file. Dunno which solution is better. I'm fine with either.

Reply:
my hunch would be that lower priority is better as it helps makes more things computable  when possible

Reply:
Neither of these two methods are working :( 

I tried `[@priority 70]` on both instances but still Lean is inferring the wrong instance. Maybe I'm doing something wrong ...

I even tried moving the `noncomputable instance` to the end of the file but then all the `linarinth` fail.

Reply:
seems `linarith` uses it in the proof...?! that is bizarre

```
@linarith.eq_of_not_lt_of_not_gt surreal
  (@linear_ordered_add_comm_group.to_linear_order surreal
    surreal.linear_ordered_add_comm_group)
```

Reply:
okay, this is incredibly confusing; you can replace the `linarith` call in `dyadic_map` (the culprit) with

```lean
      exfalso,
      subst h₂,
      exact pow_ne_zero y₃ two_ne_zero hn₃, } }
```

and then, as far as I can see, the `#print` output of `dyadic_map` and `dyadic_map._proof_1` don't mention `surreal.linear_ordered_add_comm_group` (with `pp.implicit` on), but the computer still complains I should mark it non-comp, telling me that I do indeed use it? this is truly bizarre, and may be worth asking in Zulip about, unless you have any other ideas

Reply:
Opened a new topic here: https://leanprover.zulipchat.com/#narrow/stream/144837-PR-reviews/topic/.237843.20dyadic.20surreals/near/242071240